### ISSUE TITLE
Document use cases for having Protein Complexes in GO

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-ontology#15875]
Action item from GO NYC meeting: 
Complexes working group needs to create alternate proposals for what would happen if we do/do not keep protein complexes in the CC ontology.  What are the use cases for protein-containing complexes in GO?  What would the effects be of removing them?

PIs should charge people to address question of what to do with complexes in CC - if you are interested in weighing in on this group, please add your name here: Sandra, Chris M, Helen, Judy B, Harold D, Jim Hu, Darren N, Birgit, Val, Ruth

**Comment by bmeldal** [comment]
@pgaudet people need tagging or they won't necessarily see this ticket

**Comment by pgaudet** [comment]
@cmungall @hattrill @judyblake @hdrabkin @jimhu-tamu @ValWood 
@sandraorchard

Can someone add Darren ? 

Thanks, Pascale

**Comment by cmungall** [comment]
Darren is @nataled

**Comment by bmeldal** [comment]
From call on 25th Feb 2019:

Present: Birgit, Harold, Ruth, Thomas Hayman, Kimberly, Darren, Peter, Suzie, Judy, Edith, Helen, Val

See also https://github.com/geneontology/go-ontology/issues/14375

1. All of us feel that complex terms do belong into GO as there are various use cases, including
- analysis tools
- grouping together annotations for homologous complexes
- defining intra-ontological relationships
- curation consistency

**Action 1:**
- List all use cases below (@ValWood and others)

2. To define which terms belong into GO we could utilise the logical defs (LDs) but:
- what do we do with complexes that have missing functional knowledge and can't have a LD? Will they be lost from GO? It would cause a bit loss of data.
- how would we deal with legacy terms and annotations?

NB: LDs must be **necessary and sufficient** to define a term as a unique term/class. Subclasses don't have that definition.

e.g. STING complex:
*is_a* GO:0098796 membrane protein complex
*part_of* GO:0030659 cytoplasmic vesicle membrane
*capable_of* GO:0098546 2',5-3',5'-cyclic GMP-AMP binding
*capable_of* GO:0060340 positive regulation of type I interferon-mediated signaling pathway [should it be capable_of_part_of as it's a BP ???]
*capable_of_part_of* GO:0045087 innate immune response 

These relationships probably aren't unique to the STING complex so wouldn't hold true for a LD. But people need the term to annotate the GP [Q86WV6](https://www.uniprot.org/uniprot/Q86WV6) TMEM173 (STING) to it.

Judy and Darren suggested to use PRO as it can provide LDs based on components and species but it has no integration with GO for analysis tools...

**Action:**
- continue discussion on next call.

**Comment by hattrill** [comment]
If we make a 'unique loigical definition set' rule, we need to discuss what happens to well-defined complexes that have the same logical definition - chromatin remodeling complexes spring to mind (e.g. Brahma complexes, Ino80 complex, ISWI-type complexes - all nuclear, ATPase, chromatin remodeling.

**Comment by bmeldal** [comment]
I was thinking about that last night, Helen. I've done all the SWI/SNF ones and the new GBAF is undergoing review. 

I think we'll find a few more issues... Complexes are complex ;-)

**Comment by ukemi** [comment]
If they are classes in the ontology and their logical definitions are the same, then they are the same. That's why in Birgit's notes it says that the logical definitions must be necessary and sufficient. The logical definitions have to ensure that everything that is a member of the class fits the definition and nothing out of the class fits the definition. So if a Brahma complex is defined as an 'x that relates to y' and an Ino8 complex is different but is an 'x that relates to y', then either an Ino8 complex is a Brahma complex or the definition for Brahma complex is not sufficient.  If I define a car as a 'vehicle with wheels', then a motorcycle is a car because it is a vehicle with wheels.

**Comment by hattrill** [comment]
I get that.....the logical conclusion is therefore that Brahma complexes, Ino80 complex, ISWI-type complexes should be merged into chromatin remodeling complex.....except that term has been made obsolete. Is that what we really want?

**Comment by ValWood** [comment]
Oh I missed that. Sorry my wireless connection was playing up.

So we really, really want to merge all of the chromatin remodelling complexes into "chromatin remodeling complex" . We will need alternative ways to deal with that locally. 
Personally I have published analyses where complex terms like this were useful, and our community has no other way right now to get at enrichment for complexes?

**Comment by ukemi** [comment]
I'm not saying those kinds of terms aren't useful. I'm just saying that we can't have different classes in the ontology with the same logical definitions because that means the terms are the same.

**Comment by ValWood** [comment]
I don't know what the rationale is to do this, but  the existing terms are useful:

![enrichemnt](https://user-images.githubusercontent.com/7359272/53414764-5d69ca00-39c7-11e9-8dbf-5a98fbfb2cd9.jpg)


we would need to change our
- gene pages
- query tool
- curation tool
- documentation
etc

**Comment by ValWood** [comment]
I also find these complexes to be a useful curation aid for consistency checking.

**Comment by hattrill** [comment]
One suggestion last night was that complexes should only be in GO if they could be defined by LDs....the core chromatin remodeling complexes came to mind as something that would not be able to survive this.

**Comment by hattrill** [comment]
You forgot your slim networks, Val!

**Comment by ValWood** [comment]
So I did! And will be useful in Matrix. In fact a complex is one of the examples we are using in the paper...

**Comment by bmeldal** [comment]
Chromatin remodelling complexes are really defined by their composition, and to some degree by the cell type they occur in. But that wouldn't work for GO, only PRO. And I believe enrichment tools do not integrate GO and PRO annotations.

We need a more refined solution.

**Comment by ValWood** [comment]
Can't we use PRO IDs in logical definitions for GO? We would not need to include every subunit (especially for those with compositional variation between organisms), just the 'definitive' subunits?

**Comment by ukemi** [comment]
I have always been of the opinion that if the class is useful, we should have it.

> One suggestion last night was that complexes should only be in GO if they could be defined by LDs....the core chromatin remodeling complexes came to mind as something that would not be able to survive this.

Then we need to change the rule or figure out a way to define them.

**Comment by nataled** [comment]
The other suggestion (echoing the decision from a few years ago) is that GO retain only those complexes that must be defined functionally (as opposed to by composition). If I understood the discussion yesterday, the chromatin remodeling complexes would fail that rule too. However, I also recall from that previous decision that current GO complexes stay where they are, but *new* complexes are put to whichever ontology fits the rule (GO for functional definitions, PRO for component definitions). Perhaps not the most beautiful of solutions (because of the grandfathering), but definitely an easy way to tell where something (new) should go.

**Comment by ValWood** [comment]
I was just thinking we probably could define the yeast chromatin remodeling complexes logically, with a bit of work. However, we don't actually have "chromatin remodeling complex" in GO?

**Comment by hattrill** [comment]
It was obsoleted.

**Comment by ValWood** [comment]
Sorry all this was probably discussed last night. I only caught about 50% of the conversation!

**Comment by hattrill** [comment]
No - this didn't come up last night - I only started looking this morning. Discussion centred more on things that are conserved but can't but hard to pin down exact function.

SWI/SNF superfamily-type complex is now the parent for the remodeling complexes, perhaps 'chromatin remodeling complex' was an intermediate term?

**Comment by bmeldal** [comment]
From call on 4th March:

Present: Birgit, Harold, Kimberly, Darren, Peter (left early), Edith, Suzie, Helen, Judy, Leonore, Ruth (joined later)

We recapped last week's discussion and agreed to test what would be lost of we defined GO complex terms with LDs only.

**Action:**

@nataled offered to run tests on 2 datasets:

1. Complex Portal yeast DS as it's fully annotated:
get ComplexTab file from https://www.ebi.ac.uk/complexportal/complex/organisms
GO annotation are in column H

2. a GO Plus DS

Darren will post on here when he's finished the test. No more calls set up for now.

**Comment by nataled** [comment]
Update:

I had some concerns doing the analysis on the yeast data, and now that I've done it I believe the concerns were warranted. It was fairly easy to do though, so...

We were interested in a quick assessment of how many complexes in GO can be uniquely defined using "necessary and sufficient" logical definitions. The proposal for this test is based on the idea that complexes with a unique combination of annotations might represent low-hanging fruit, if the annotations were converted to components of the definition. As a first pass, it was suggested that the complexes described for S. cerevisiae be tested. I've had a look at the data, assessing the complexes named in column H as the complex of consideration. For example, CPX-3248 (Asi complex; line 366 of the tsv) has the following:

  GO:0097658(Asi complex)
  GO:0004842(ubiquitin-protein transferase activity)
  GO:0005637(nuclear inner membrane)
  GO:0006511(ubiquitin-dependent protein catabolic process)
  GO:0071230(cellular response to amino acid stimulus)

The assumption is that the complex of interest here is the species-generic GO:0097658 (Asi complex), and that it could potentially be logically defined as:

  intersection_of: GO:0032991 ! protein-containing complex
  intersection_of: capable_of GO:0004842 ! ubiquitin-protein transferase activity
  intersection_of: localized_to GO:0005637 ! nuclear inner membrane
  intersection_of: involved_in GO:0006511 ! ubiquitin-dependent protein catabolic process
  intersection_of: involved_in GO:0071230 ! cellular response to amino acid stimulus

(the precise and correct relations are immaterial for this analysis, so pay no mind if they are wrong)

Conversely, CPX-1355 (RTT107-SLX4-SLX1 complex; line 2 of the tsv file) lacks a protein complex parent:
  GO:2000001(regulation of DNA damage checkpoint)
  GO:0005634(nucleus)

Accordingly, CPX-1355 cannot be used for this analysis. 

Some yeast complexes mention multiple potential parent terms. Likely these can be figured out semi-automatically, but for now I ignored them since they complicate and slow the analysis. An example of this is CPX-3265 (RQC complex, line 343):

  GO:1990116(ribosome-associated ubiquitin-dependent protein catabolic process)
  GO:1990112*(RQC complex)
  GO:0022625*(cytosolic large ribosomal subunit)
  GO:0061630(ubiquitin protein ligase activity)

Finally, there are cases where the complex of interest appears multiple times. Most often the potential logical definitions (based on the information in column H) are in accordance with each other and therefore are okay to use. For example, though GO:0031931 "TORC1 complex" appears as parent for both CPX-1715 (TORC1 complex variant 1; line 204) and CPX-1716 (TORC1 complex variant 2; line 196), both lines are in agreement about how the parent term would be defined. However, sometimes this is not the case. For example, GO:1905348 "endonuclease complex" appears as parent for both CPX-1362 (SLX4-RAD1-RAD10 endonuclease complex; line 3) and CPX-1363 (SAW1-RAD1-RAD10 endonuclease complex; line 4). Each of the column H annotations slightly differ. Both contain the following:

  GO:1905348(endonuclease complex)
  GO:0004519(endonuclease activity)
  GO:0000736(double-strand break repair via single-strand annealing, removal of nonhomologous ends)
  GO:0070337(3'-flap-structured DNA binding)

But CPX-1363, in addition, specifies:
  GO:0005634(nucleus)

This difference makes the final potential logical definition unclear, so these too were ignored.

That brings us to the results. Of the 589 complexes in the file, 60 were ignored because there was no parent complex indicated, and 28 were ignored because there were multiple parent complexes indicated. A large number (110) were ignored because the indicated parent complex was found multiple times in the file and had different annotations associated with it. That leaves a total of 391 complexes usable for this analysis.

Of the 391, 314 (>80%) would be defined uniquely, purely on the basis of GO MF or CC or BP terms. There are two important caveats here:

1) The assumption is that the annotations given within column H represent the actual annotations of the parent complex. The fact that there are 110 cases where the parent complex has different annotations indicates that this assumption is incorrect.

2) It is quite likely that a similar analysis using a different or additional file would give different results.

To be honest, I don't trust these results to tell us anything useful. We need the actual annotations used for the complexes. Presumably these can be found in GO-plus.

**Comment by bmeldal** [comment]
Hi Darren, 

Thank you for running this analysis, it is showing exactly what I expected, comments in line:

>I had some concerns doing the analysis on the yeast data, and now that I've done it I believe the concerns were warranted. It was fairly easy to do though, so...

>We were interested in a quick assessment of how many complexes in GO can be uniquely defined using "necessary and sufficient" logical definitions. The proposal for this test is based on the idea that complexes with a unique combination of annotations might represent low-hanging fruit, if the annotations were converted to components of the definition. As a first pass, it was suggested that the complexes described for S. cerevisiae be tested. I've had a look at the data, assessing the complexes named in column H as the complex of consideration. For example, CPX-3248 (Asi complex; line 366 of the tsv) has the following:

>GO:0097658(Asi complex)
GO:0004842(ubiquitin-protein transferase activity)
GO:0005637(nuclear inner membrane)
GO:0006511(ubiquitin-dependent protein catabolic process)
GO:0071230(cellular response to amino acid stimulus)

>The assumption is that the complex of interest here is the species-generic GO:0097658 (Asi complex), and that it could potentially be logically defined as:

>intersection_of: GO:0032991 ! protein-containing complex
intersection_of: capable_of GO:0004842 ! ubiquitin-protein transferase activity
intersection_of: localized_to GO:0005637 ! nuclear inner membrane
intersection_of: involved_in GO:0006511 ! ubiquitin-dependent protein catabolic process
intersection_of: involved_in GO:0071230 ! cellular response to amino acid stimulus

>(the precise and correct relations are immaterial for this analysis, so pay no mind if they are wrong)

>Conversely, CPX-1355 (RTT107-SLX4-SLX1 complex; line 2 of the tsv file) lacks a protein complex parent:
GO:2000001(regulation of DNA damage checkpoint)
GO:0005634(nucleus)

>Accordingly, CPX-1355 cannot be used for this analysis.

Correct, this is due to a different curation strategy at the start of the project and complex annotations could be added retrospectively. (We [= I!] would have to find time for this)

>Some yeast complexes mention multiple potential parent terms. Likely these can be figured out semi-automatically, but for now I ignored them since they complicate and slow the analysis. An example of this is CPX-3265 (RQC complex, line 343):

>GO:1990116(ribosome-associated ubiquitin-dependent protein catabolic process)
GO:1990112*(RQC complex)
GO:0022625*(cytosolic large ribosomal subunit)
GO:0061630(ubiquitin protein ligase activity)

This is a mistake as RQC can't be associated with the 60S and part of it at the same time. I've fixed it and the update will be out with the next release - nice spot!  **- Hey, and a use case for complexes in GO ;-)**

But, of course, there might be other examples where a complex IS part of 2 different parents as complexes, just like proteins, are capable of moonlighting :) I would be interested in the list of these (you can send them directly if you have the list as part of your results) as it might be that there are more mistakes.

Another odd case is:
CPX-1891 RPB4-RPB7 subcomplex
is_a GO:1990328 RPB4-RPB7 complex --> parent GO:0032991 protein-containing complex
part_of GO:0000932 P-body
part_of GO:0005665 RNA polymerase II, core complex 

Here, the problem lies in the fact that in the CP editor we cannot assign the annotation relationships and we have to export the default ones (part_of) but the complex exists on its own as well as subcomplex of the RNAPII core. But maybe there aren't many of these and they could be "rescued" by hand.

>Finally, there are cases where the complex of interest appears multiple times. Most often the potential logical definitions (based on the information in column H) are in accordance with each other and therefore are okay to use. For example, though GO:0031931 "TORC1 complex" appears as parent for both CPX-1715 (TORC1 complex variant 1; line 204) and CPX-1716 (TORC1 complex variant 2; line 196), both lines are in agreement about how the parent term would be defined. However, sometimes this is not the case. For example, GO:1905348 "endonuclease complex" appears as parent for both CPX-1362 (SLX4-RAD1-RAD10 endonuclease complex; line 3) and CPX-1363 (SAW1-RAD1-RAD10 endonuclease complex; line 4). Each of the column H annotations slightly differ. Both contain the following:

>GO:1905348(endonuclease complex)
GO:0004519(endonuclease activity)
GO:0000736(double-strand break repair via single-strand annealing, removal of nonhomologous ends)
GO:0070337(3'-flap-structured DNA binding)

>But CPX-1363, in addition, specifies:
GO:0005634(nucleus)

>This difference makes the final potential logical definition unclear, so these too were ignored.

Again, this is because we didn't systematcally add the cell_part to each complex. This can be fixed (with some magic time!) and then wouldn't break the pipeline.

>That brings us to the results. Of the 589 complexes in the file, 60 were ignored because there was no parent complex indicated, and 28 were ignored because there were multiple parent complexes indicated. A large number (110) were ignored because the indicated parent complex was found multiple times in the file and had different annotations associated with it. That leaves a total of 391 complexes usable for this analysis.

>Of the 391, 314 (>80%) would be defined uniquely, purely on the basis of GO MF or CC or BP terms. There are two important caveats here:

> - The assumption is that the annotations given within column H represent the actual annotations of the parent complex. The fact that there are 110 cases where the parent complex has different annotations indicates that this assumption is incorrect.

I think there are missing annotations due to an evolving curation practice (pilot vs fully-fletched GO annotations)

> - It is quite likely that a similar analysis using a different or additional file would give different results.

Need to test that.

>To be honest, I don't trust these results to tell us anything useful. We need the actual annotations used for the complexes. Presumably these can be found in GO-plus.

No, they are not in GO plus as many of the annotations in the ComplexTab file have no fully-annotated GO lines (historic annotations) so are excluded from the GPAD export and are only stored in our DB. 

**What this exercise has shown:**

- I should check the 28 complexes with multiple parents - QC exercise
- >80% of complexes have **necessary and sufficient** annotations to make them unique. The excluded 110 + 60 complexes will have a similar ratio or be less unique (I know a bit more about these subsets as I worked on them).

--> That means about 20% of all yeast complexes can **NOT be defined as necessary and sufficient** by their annotations. That's a fairly large number and is exactly what I expected.

--> It won't be trivial to define complexes purely by their LDs.

NB: as we add more and more human complexes we will add more variants that are identical except for their composition.

**Comment by ValWood** [comment]
I think it's pretty impressive that 80% are already defined uniquely !

**Comment by bmeldal** [comment]
From Cambridge GOC meeting, 11/4/19:

- [ ] @bmeldal will look at the 20% of CP complexes that can’t be defined logically. 
- [ ] Could we define more complexes by using more granular MF terms? Or by using other ontologies, e.g. cell type ontology?
- [ ] @bmeldal to invite Colin Logie to relevant meetings/TCs
- [ ] @vanaukenk / @ukemi / @hdrabkin ? Convert all subclass definitions to logical definitions where possible

**Comment by pgaudet** [comment]
> Could we define more complexes by using more granular MF terms? Or by using other ontologies, e.g. cell type ontology?

I'll add this to the next Ontology call (I dont see why not!) 

Thanks, Pascale

**Comment by bmeldal** [comment]
If agreed in principle, which I thought we had last week, I think we need to define this a bit more once I have looked at those "20% yeast complexes" and have some examples.

**Comment by pgaudet** [comment]
@bmeldal Do you have examples for terms for which you would need external ontologies to create a unique definition?

**Comment by bmeldal** [comment]
[GO:0070603 SWI/SNF superfamily-type complex](https://www.ebi.ac.uk/QuickGO/term/GO:0070603)

They are all ATPases acting as chromatin remodellers and are distinguished by their composition based on cell/tissue type and developmental state.

**Comment by pgaudet** [comment]
Thanks
@vanaukenk  @ukemi based on our discussion on yesterda'y ontology call, would you suggest merging those terms? 

Thanks, Pascale

**Comment by bmeldal** [comment]
Merging all SWI/SNF complexes into one would loose a lot of knowledge.

**Comment by bmeldal** [comment]
@colinlog what do you think as end user and SWI/SNF expert?

**Comment by colinlog** [comment]
Hi, I am (still) a GO ignoramus: what is an LD? 
Also 'necessary and sufficient' in the ontological sense is unclear to me. 
I need to understand what you are talking about before I comment on the June 6, 2018 part of this ticket.
 
What I can say as end-user is that I would like all the SWI/SNF complexes to fall under one parent term that is called SWI/SNF with synonyms such as BAF, P/BAF etc.) The justification for this is 
(1) that they all share one or two highly related ATPases that can be distinguished from the other SNF2-type ATPases like INO80 or ISWI, eg: they have a C-terminal Bromodomain). At the same ontologyevel as SWI/SNF complexes I would like to see ISWI, INO80, SWR1, ... . So a generic name for a complex that may have different compositions in one cell type, between cell types and between organisms. Creating that parent is not a merge, it would be an 'umbrella' and would be useful in many ways, including the future annotation of the Sea Horse genome and some such.
(2) that a complex that is only p[artially known can be seen as just that. the complex does not need to be fully elucidated to be part of the annotated complex collection.

I think that generating all the combinatorially described complexes should be done operationally rather than as a set of 'flatfiles' as I saw in complex portal. The large number of combinations is rather overwhelming... and appears to occlude existing knowledge. Furthermore, to be useful and more future proof, addressing complex isoforms could be potentially made depend on the cellular context (transcriptome, proteome) such that one gets to see/select the presence of any subunit paralogue in the cell type that one is considering.

The complex portal should be easy to navigate, and one way of doing that is to have a bit more of a hierarchy, perhaps?

I would like that 'clicking' on SWI/SNF complex 'super-family' would list all the (human) sub-units that can be part of the set of (human SWI/SNF) complexes rather than all the possible complexes.  Then, in parralllel I wish to be able to access the known complex variants with their proteomic/ontology attributes. Eg: SWI/SNF with Polybromo and ARID2 is P/BAF, without polybromo and ARID2 it is BAF, with BAF53B it is called nBAF, etc...) 
At the level of subunits, I would like to be able to visualise/list/access the mutually exclusive subunits as such: BAF can exist with DPF1, DPF2 or DPF3, but not any two together. Those are mutually exclusive subunits). And so on... 

What I am arguing for is that listing all the combinations without overarching logic is not very useful and potentially generates much repetition, failing to highlight what makes any complex variant what it is. If there is no extra knowledge available as to complex variants then list all the subunits and we can figure out experimentally or through literature mining whether there are different variants of that complex.

**Comment by deustp01** [comment]
LD = logical definition

**Comment by bmeldal** [comment]
Colin, I'll answer at length but maybe not til tomorrow as finishing a report that's 3 month overdue :(

**Comment by colinlog** [comment]
Hi Brigit, I will be in Heidelberg the rest of the week, so it may be slow
from my side. I have not yet made true my promise to send you a plan of
what I want to do and in what order I think that makes most sense. It may
be some weeks still. Meanwhile, interacting via GitHub with you and the GO
consortium most definitely inspires me! Best, Colin

**Comment by ValWood** [comment]
I like the current specificity.  We use it to group like this:
https://www.pombase.org/term/GO:0070603

At this level are all descendant complexes defined uniquely (albeit not in a way that could be defined logically because the definitions are based on protein family information).   Instead, each complex lists some unique aspect of the composition to make them mutually exclusive.

This is useful because even the people who work on these can't hold this info in their heads.
What is the issue here? I'm not following?

I remember @colinlog  saying recently that these could be defined with respect to their specific nucelotide modelling role (i.e which nucleosome/histone or position they act upon IIRC?). This could be made compatible with logical definitions, and seems to be important information about the functions that we would want to capture. It should provide a way to ensure these complexes are annotated to the correct aspects of transcription and/or chromatin remodelling.

**Comment by ValWood** [comment]
I just re-read the thread. I still don't understand fully the issue.

80% are defined uniquely.
Why not get a list and just create unique defs for the ones which don't have them and merge any redundency?

**Comment by pgaudet** [comment]
> Why not get a list and just create unique defs for the ones which don't have them and merge any redundency?

That is probably the best solution, unless some terms are absolutely needed - @bmeldal is this the case ?

**Comment by bmeldal** [comment]
Hi all, 

I will answer Colin's great input step by step.

>What is an LD?
>Also 'necessary and sufficient' in the ontological sense is unclear to me.

**Logical definitions (LDs):**

define a term as unique (= necessary and sufficient), using relationships to other GO terms,

e.g. 
GO:1904949 ATPase complex 
enables GO:0016887 ATPase activity
= some complexes that has ATPase activity is an ATPase complex
or
GO:1902495 transmembrane transporter complex
enables GO:0022857 transmembrane transporter activity 
AND 
part_of GO:0016021 integral component of membrane
= some complex that is an integral component of some membrane AND has transmembrane transporter activity is a transmembrane transporter complex

We are thereby defining these terms by what they do in terms of MF/BP/CC (as applicable) and delineate them from each other by their logical definitions. 
But often terms are not define uniquely, because
a) we simply didn't apply LDs systematically (main cases I suspect!)
b) because we don't have the necessary knowledge
c) we haven't got the right terms in the ontology (which can, of course, be added!).

One example we discussed are the SWI/SNF complexes:
GO:0070603 SWI/SNF superfamily-type complex
part_of GO:0000790 nuclear chromatin
is_a GO:1904949 ATPase complex
   --> enables GO:0016887 ATPase activity (MF via its CC parent)

As you can see, SWI/SNF superfamily-type complex is distinguished from ATPase complex by having an additional relationship to nuclear chromatin.

But some of its children, e.g.
GO:0016514 SWI/SNF complex
and
GO:0035060 brahma complex
have the same relationships as each other, so cannot be uniquely separated:
is_a GO:0070603 SWI/SNF superfamily-type complex
   --> enables GO:0016887 ATPase activity (via parents)
   --> part_of GO:0000790 nuclear chromatin (via parents)
enables GO:0003690 double-stranded DNA binding
enables GO:0043044 ATP-dependent chromatin remodeling
involved_in GO:0006357 regulation of transcription by RNA polymerase II

while other children are only is_a children of GO:0070603 SWI/SNF superfamily-type complex - which doesn't differentiate them from their parent terms AND is rather inconsistent compared to its siblings (lack of an LDs added)...

- Overall, it means that none of the existing SWI/SNF complexes and they children have been uniquely defined yet. 
- At the GOC mtg we discussed the possibility of creating more granular terms to delineate these complexes better and with proper LDs.
- We also suggested that we could use terms from other ontologies (PRO, SO...) to make the LDs for complexes.


**Umbrella/Parent terms for complex families:**

>What I can say as end-user is that I would like all the SWI/SNF complexes to fall under one parent term that is called SWI/SNF with synonyms such as BAF, P/BAF etc.) The justification for this is
(1) that they all share one or two highly related ATPases that can be distinguished from the other SNF2-type ATPases like INO80 or ISWI, eg: they have a C-terminal Bromodomain). At the same ontology level as SWI/SNF complexes I would like to see ISWI, INO80, SWR1, ... . So a generic name for a complex that may have different compositions in one cell type, between cell types and between organisms. Creating that parent is not a merge, it would be an 'umbrella' and would be useful in many ways, including the future annotation of the Sea Horse genome and some such.

Questions: 
1. One parent term in GO or CP? Cos in GO it is almost like that, except, currently, the parent term includes all SWI/SNF-type complexes and the different BAFs (SWI/SNF, pBAF, GBAF...) and other chromatin remodelers (INO80, ISWI complex...) are all siblings.
2. Would you want to retain the current BAF children in GO or prefer to ONLY have the parent term (that doesn't exist yet)?
NB: We had a lengthy [discussion](https://github.com/geneontology/go-ontology/issues/14106) on BAFs 2 years ago and ended up merging the previous BAF parent (GO:0090544 BAF-type complex) and it's siblings into GO:0070603 SWI/SNF superfamily-type complex (see comment from 4/9/17).

>(2) that a complex that is only partially known can be seen as just that. The complex does not need to be fully elucidated to be part of the annotated complex collection.

Neither resource will create terms for partially-defined complexes, and I don't think, that's what you are asking for. What search result do you expect if your query is an incomplete protein list? At the moment, in either resource, it would suggest any complexes that have these members as components (CP) or proteins that are already annotated to this term (GO) but wouldn't tell you how good/complete your coverage is.

>I think that generating all the combinatorially described complexes should be done operationally rather than as a set of 'flatfiles' as I saw in complex portal. The large number of combinations is rather overwhelming... and appears to occlude existing knowledge. Furthermore, to be useful and more future proof, addressing complex isoforms could be potentially made depend on the cellular context (transcriptome, proteome) such that one gets to see/select the presence of any subunit paralogue in the cell type that one is considering.

This really is a question for us at CP and we'll discuss it off-ticket. It is very philosophical in the first place, and once we have a general solution it will need quite a bit of development to implement it.

>The complex portal should be easy to navigate, and one way of doing that is to have a bit more of a hierarchy, perhaps?

Currently, we don't provide ANY hierarchy in the CP. We had discussions with PRO about it but it hasn't been implemented yet. Again, I think we'll discuss options off-ticket initially.

>I would like that 'clicking' on SWI/SNF complex 'super-family' would list all the (human) sub-units that can be part of the set of (human SWI/SNF) complexes rather than all the possible complexes. Then, in parallel I wish to be able to access the known complex variants with their proteomic/ontology attributes. Eg: SWI/SNF with Polybromo and ARID2 is P/BAF, without polybromo and ARID2 it is BAF, with BAF53B it is called nBAF, etc...)
At the level of subunits, I would like to be able to visualise/list/access the mutually exclusive subunits as such: BAF can exist with DPF1, DPF2 or DPF3, but not any two together. Those are mutually exclusive subunits). And so on...
What I am arguing for is that listing all the combinations without overarching logic is not very useful and potentially generates much repetition, failing to highlight what makes any complex variant what it is. If there is no extra knowledge available as to complex variants then list all the subunits and we can figure out experimentally or through literature mining whether there are different variants of that complex.

A very good point but CP-related. This implies using something like "sets" in the CP and requires more development. It is something we are discussing in the team right now.

I hope I addressed all points for now.
